### DISCUSSION TITLE
macOS build - [74%] ERROR: Illegal package file name...

### DISCUSSION THREAD

**Discussion by 17matejka** [AcademySoftwareFoundation/OpenRV#85] [Build Issues]
I am trying to build Open RV on macOS Big Sur 11.7.4 with Xcode 13.2.1.
During process I saw some warnings, but no error I believe.
When process is in: **[ 74%] Installing Open RV Packages**
This error pops up:
```
[ 74%] Installing Open RV Packages
INFO: package failed to copy: ERROR: Illegal package file name: additional_nodes-.rvpkg, should be either <name>.zip or 
<name>-<version>.rvpkg
exiting
make[3]: *** [stage/app/RV.app/Contents/PlugIns/Packages/rvinstall] Error 255
make[2]: *** [src/bin/apps/rvpkg/CMakeFiles/rvinstall.dir/all] Error 2
make[1]: *** [src/bin/nsapps/RV/CMakeFiles/RV.dir/rule] Error 2
make: *** [RV] Error 2
```

I tried to modify `../OpenRV/cmake/macros/rv_stage.cmake` and `../OpenRV/cmake-build/src/bin/apps/rvpkg/CMakeFiles/rvinstall.dir/build.make` to make pkg name not ending with "...-" e.g. additional_nodes-v001.rvpkg but the error is still the same.

Any ideas what is going on?

**Comment by rogernelson** [comment]
Hi @17matejka.  The part you're trying to remove is the package version.  I'd need to look into it further to know for sure why it complains about that, but my guess is that RV wants to be able to determine the version of the package without needing to open it first. Is there a particular reason you are trying to remove it?

**Reply by 17matejka** [reply]
Hi @rogernelson There is no particular reason, I was only trying to debug it with my poor dev experience. My idea was manually force version into package name, because obviously version is missing in the naming of the packages, but that did not work

**Reply by rogernelson** [reply]
Have you pushed your changes somewhere where I can take a look? I might be able to spot what is going wrong.

**Reply by 17matejka** [reply]
No, after messing around with it for a while I reverted all my changes (the original error msg is from default). But basically I was going through .cmake, .make and Makefile which were looking somewhat linked to rv packages. In those files I was looking for phrase ".rvpkg" and then I was changing default version variables (or non values in strings - package names) to v001 (it was in rv_stage.cmake I believe, sorry I do not have my business laptop at home with me) to see what it does. The error illegal package file name continued even if package was named e.g. additional_nodes-v001.rvpkg or additionalnodes.rvpkg

In one of the .cmake file (probably rv_stage.cmake) there are lines, that should generate package name with version in this format package_name-version. Name and version are defined by 2 separate variables and it looks like that version variable is empty, so that is the reason why packges are named e.g. additional_nodes-.rvpkg

I would like to help you more with this, but I just started playing with this week ago (as non-developer) on Windows and after successful build I switched to macOS.

**Reply by rogernelson** [reply]
Yes, if a CMake variable is unset it will default to an empty string.  So if you make a typo in the variable name or use the wrong syntax for a variable in a string (it should look like `${MY_VAR_NAME}` then you can end up with empty strings where you didn't intend.  It may have been something like that that happened to you.

If you take another shot at it, you can paste the link to the source code, and I'll take a look at your changes and see if I can help.

**Reply by 17matejka** [reply]
Got it, but sounds like misunderstanding, I probably did not discribe the problem properly.
That error pops up without any my changes, clean cloned repo, followed macOS instructions. 
After it showed up I tried to fixed it with modifying cmake...
